docs/elements/guides/styling.mdx

```diff
@@ -0,0 +1,270 @@
+          <Clerk.Input className={styles.input} />
+          <Clerk.FieldError className={styles.error} />
+        </Clerk.Field>
+        <SignIn.Action submit className={styles.submit}>Continue</SignIn.Action>
+      </SignIn.Step>
+      <SignIn.Step name="verifications" className={styles.verificationsStep}>
+        <SignIn.Strategy name="email_code">
+          <Clerk.Field name="code">
+            <Clerk.Label className={styles.label}>Code</Clerk.Label>
+            <Clerk.Input className={styles.input} />
+            <Clerk.FieldError className={styles.error} />
+          </Clerk.Field>
+          <SignIn.Action submit className={styles.submit}>Verify</SignIn.Action>
+      </SignIn.Step>
+    </SignIn.Root>
+  )
+}
+```
+
+### Inline styles
+
+Just like native HTML elements, you can also use inline styles to style Clerk Elements. This is useful when you need to apply styles conditionally or when you want to avoid creating a separate CSS file.
+
+```tsx filename="/app/sign-in/[[...sign-in]]/page.tsx"
+import * as Clerk from '@clerk/elements/common'
+import * as SignIn from '@clerk/elements/sign-in'
+
+export default function SignInPage() {
+  return (
+    <SignIn.Root>
+      <SignIn.Step name="start" style={{
+        backgroundColor: 'white',
+        padding: '1rem',
+        border: '1px solid #dddddd',
+        borderRadius: '4px',
+      }}>
+        <Clerk.Field name="identifier">
+          <Clerk.Label style={{
+            color: '#dddddd',
+            fontSize: '0.875rem',
+          }}>Email</Clerk.Label>
+          <Clerk.Input style={{
+            border: '1px solid #dddddd',
+            borderRadius: '4px',
+            padding: '1rem',
+          }} />
+        </Clerk.Field>
+        <SignIn.Action submit style={{
+          backgroundColor: '#111111',
+          color: 'white',
+          padding: '1rem',
+          borderRadius: '4px',
+        }}>Continue</SignIn.Action>
+      </SignIn.Step>
+    </SignIn.Root>
+  )
+}
+```
+
+## State-based styling
+
+In some cases you might want to style components based on their state. Clerk Elements exposes data attributes for this purpose, as well as components that expose the state programmatically to support more complex logic.
+
+### Data attributes
+
+#### Field components
+
+`<Field>` and `<Input>` can be styled based on their validity state by targeting the `data-valid` or `data-invalid` attributes:
+
+```css filename="style.css"
+.input {
+  --border-color: gray;
+  border: 1px solid var(--border-color);
+
+  &[data-invalid] {
+    --border-color: red;
+  }
+}
+```
+
+```tsx filename="page.tsx"
+<Clerk.Input className="input" />
+```
+
+### Function as children
+
+If you need programmatic access to state for more complex styling, several components provide a function as children. This is useful when dealing with animations, or for conditionally rendering elements based on state.
+
+For example: To access a field's state, use `<FieldState>`:
```

Comment:
Are there other options like this? LabelState for example? Or is it just FieldState?

I ask because the text in this section make it feel like there are more options, but all I could find in the PR after a quick look was `FieldState`

Reply:
It's just `<FieldState>`. So you can use `<FieldState>` to access the state of a `<Field>`. But `<Field>` also accepts a function as `children` which you can use to access it, too.

I think we can just remove the confusion here by not mentioning it that there are more "options"